### PR TITLE
Learn js basics

### PR DESCRIPTION
**Pull request by Maxime-Simon**

### CODE REVIEW

File: javascripting/acces-valeur-tableau.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const food = ['apple', 'pizza', 'pear']
+console.log(food[2 - 1])
```
Comment (jmisiti42) [inline javascripting/acces-valeur-tableau.js]: You should add a `newline` at the end of your files. It is a common norm

File: javascripting/arguments-de-fonction.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+function math(x, y, z){
+    return ( y * z ) + x
```
Comment (jmisiti42) [inline javascripting/arguments-de-fonction.js]: No space after and before parentheses.
 `return (y * z) + x`

File: javascripting/boucle-for.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+let total = 0
+const limit = 10
+
+for(let i = 0; i < limit; i++){
+    total += i
+}
+console.log(total)
```
Comment (jmisiti42) [inline javascripting/boucle-for.js]: You could'v do like this:
`let i = 0;`
`for (i; i < 5; i++) {}`
`console.log(i)`

File: javascripting/iterer-dans-des-tableaux.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+let pets = ['cat', 'dog', 'rat']
+
+for(let i = 0 ; i < pets.length;  i++){
+    pets[i] = `${pets[i]}s`
```
Comment (jmisiti42) [inline javascripting/iterer-dans-des-tableaux.js]: This is swag 👍

File: javascripting/boucle-for.js (hunk 3 above)
Comment (Maxime-Simon) [reply javascripting/boucle-for.js]: i don't think so.

File: javascripting/objets.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const pizza = {
+    toppings: ['cheese', 'sauce', 'pepperoni'],
+    crust: 'deep dish',
+    serves: 2
+  }
+  console.log(pizza)
```
Comment (jmisiti42) [inline javascripting/objets.js]: This indentation sucks

File: javascripting/program.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+const example = 'example string'
+console.log (example.length)
```
Comment (jmisiti42) [inline javascripting/program.js]: Don't add spaces before parentheses 
`console.log(example.length)`

File: javascripting/proprietes-objet.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const food = {
+    types: 'only pizza'
+  }
```
Comment (jmisiti42) [inline javascripting/proprietes-objet.js]: This `}` should not be indented

File: javascripting/revisions-chaines.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const pizza = 'pizza is alright'
+const ret = pizza.replace('alright', 'wonderful')
+     console.log(ret)
```
Comment (jmisiti42) [inline javascripting/revisions-chaines.js]: Wrong indentation

File: javascripting/arguments-de-fonction.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 function math(x, y, z){
-    return ( y * z ) + x
+    return (y * z) + x
 }
 
 console.log(math(53, 61, 67))
```
Comment (jmisiti42) [inline javascripting/arguments-de-fonction.js]: newline

File: javascripting/objets.js [hunk 10]
```diff
@@ -2,5 +2,5 @@ const pizza = {
     toppings: ['cheese', 'sauce', 'pepperoni'],
     crust: 'deep dish',
     serves: 2
-  }
-  console.log(pizza)
\ No newline at end of file
+}
+console.log(pizza)
```
Comment (jmisiti42) [inline javascripting/objets.js]: newline

File: javascripting/program.js [hunk 11]
```diff
@@ -1,2 +1,2 @@
 const example = 'example string'
-console.log (example.length)
\ No newline at end of file
+console.log(example.length)
```
Comment (jmisiti42) [inline javascripting/program.js]: newline

File: javascripting/revisions-chaines.js [hunk 12]
```diff
@@ -1,3 +1,3 @@
 const pizza = 'pizza is alright'
 const ret = pizza.replace('alright', 'wonderful')
-     console.log(ret)
\ No newline at end of file
+console.log(ret)
```
Comment (jmisiti42) [inline javascripting/revisions-chaines.js]: new line

File: javascripting/chaines.js [hunk 13]
```diff
@@ -0,0 +1,3 @@
+const someString = 'this is a string'
+
+console.log(someString)
```
Comment (jmisiti42) [inline javascripting/chaines.js]: newline

File: javascripting/filtrage-de-tableaux.js [hunk 14]
```diff
@@ -0,0 +1,4 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+const filtered = numbers.filter(num => num % 2 === 0)
+console.log(filtered)
```
Comment (jmisiti42) [inline javascripting/filtrage-de-tableaux.js]: newline

File: javascripting/functions.js [hunk 15]
```diff
@@ -0,0 +1,5 @@
+function eat (food) {
+   return `${food} tasted really good.`
+}
+
+console.log(eat('bananas'))
```
Comment (jmisiti42) [inline javascripting/functions.js]: nl

File: javascripting/instruction-conditionnelle.js [hunk 16]
```diff
@@ -0,0 +1,8 @@
+const fruit = "orange"
+
+if(fruit.length > 5){
+    console.log("The fruit name has more than five characters.")
+}
+else {
+    console.log("The fruit name has five characters or less")
+}
```
Comment (jmisiti42) [inline javascripting/instruction-conditionnelle.js]: nl

File: javascripting/introduction.js [hunk 17]
```diff
@@ -0,0 +1 @@
+console.log('hello')
```
Comment (jmisiti42) [inline javascripting/introduction.js]: nl

File: javascripting/iterer-dans-des-tableaux.js [hunk 18]
```diff
@@ -0,0 +1,7 @@
...
+}
+
+console.log(pets)
```
Comment (jmisiti42) [inline javascripting/iterer-dans-des-tableaux.js]: tmtc

File: javascripting/nombres-arrondis.js [hunk 19]
```diff
@@ -0,0 +1,3 @@
+const roundUp = 1.5
+const rounded = Math.round(roundUp)
+console.log(rounded)
```
Comment (jmisiti42) [inline javascripting/nombres-arrondis.js]: tmtc

File: javascripting/nombres-en-chaines.js [hunk 20]
```diff
@@ -0,0 +1,3 @@
+let n = 128
+n = n.toString()
+console.log(n)
```
Comment (jmisiti42) [inline javascripting/nombres-en-chaines.js]: tmtc

File: javascripting/nombres.js [hunk 21]
```diff
@@ -0,0 +1,2 @@
+const nbr = 123456789
+console.log(nbr)
```
Comment (jmisiti42) [inline javascripting/nombres.js]: tmtc

File: javascripting/objects-keys.js [hunk 22]
```diff
@@ -0,0 +1,9 @@
+const car = {
+    make: 'Honda',
+    model: 'Accord',
+    year: 2020
+}
+
+const keys = Object.keys(car)
+
+console.log(keys)
```
Comment (jmisiti42) [inline javascripting/objects-keys.js]: tmtc

File: javascripting/proprietes-objet.js [hunk 23]
```diff
@@ -0,0 +1,6 @@
...
+const key = "types"
+
+console.log(food[key])
```
Comment (jmisiti42) [inline javascripting/proprietes-objet.js]: tmtc

File: javascripting/tableaux.js [hunk 24]
```diff
@@ -0,0 +1,2 @@
+const pizzaToppings = ['tomato sauce', 'cheese', 'pepperoni']
+console.log(pizzaToppings)
```
Comment (jmisiti42) [inline javascripting/tableaux.js]: tmtc

File: javascripting/variables.js [hunk 25]
```diff
@@ -0,0 +1,2 @@
+const example = "some string"
+console.log(example)
```
Comment (jmisiti42) [inline javascripting/variables.js]: tmtc

**Pull request merged**